parachain/primitives/ethereum/src/lib.rs

```diff
@@ -99,6 +102,32 @@ impl Header {
 		keccak_256(&self.rlp(false)).into()
 	}
 
+	pub fn check_receipt_proof(&self, proof: &[Vec<u8>]) -> Option<Receipt> {
+		match self.apply_merkle_proof(proof) {
+			Some((root, data)) if root == self.receipts_root => rlp::decode(&data).ok(),
+			Some((_, _)) => None,
+			None => None,
+		}
+	}
+
+	pub fn apply_merkle_proof(&self, proof: &[Vec<u8>]) -> Option<(H256, Vec<u8>)> {
+		let mut iter = proof.iter().rev();
```

Comment:
If you're consuming `proof` and don't intend to use it again, consider using `into_iter()` ([IntoIter](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html#tymethod.into_iter)), then you can work with values directly instead of references to values.